msal/application.py

```diff
@@ -1702,6 +1702,21 @@ def acquire_token_for_client(self, scopes, claims_challenge=None, **kwargs):
             - an error response would contain "error" and usually "error_description".
         """
         # TBD: force_refresh behavior
+        if self.client_credential is None:
```

Comment:
I think you should have ManagedIdentity of something like that, because in the future we'll have the certificateless ones (several type of not client credentials). You could also have Pod Identity when deployed in a container in AKS